docs/docs/guides/asynchronous-state-sync.md

```diff
@@ -2,3 +2,185 @@
+
+## Local State Example
+
+This example provides the friend status as local state only.
+
+```jsx
+const currentUserIDState = atom({
+  key: 'CurrentUserID',
+  default: null,
+});
+
+function CurrentUserInfo() {
+  const [currentUserID] = useRecoilState(currentUserIDState);
+  return <div>Current User: {currentUserID}</div>;
+}
+```
+
+## Sync State From Server
+
+We can subscribe to asynchronous changes in the remote state and update the atom value to match.  This can be done using standard React `useEffect()` hook or other popular libraries.
+
+```jsx
+function CurrentUserIDSubscription() {
+  const setCurrentUserID = useSetRecoilState(currentUserIDState);
+
+  useEffect(() => {
+    RemoteStateAPI.subscribeToCurrentUserID(setCurrentUserID);
+    // Specify how to cleanup after this effect
+    return function cleanup() {
+      RemoteServerAPI.unsubscribeFromFriendStatus(setCurrentUserID);
+    };
+  }, []);
+
+  return null;
+}
+
+function MyApp() {
+  return (
+    <RecoilRoot>
+      <CurrentUserIDSubscription />
+      <CurrentUserInfo />
+    </RecoilRoot>
+  );
+}
+```
+
+If you want to handle synchronization of multiple atoms in a single place, you can also use the [State Persistence](persistence) pattern.
+
+## Bi-Directional Synching
+
+You can also sync the state so local changes are updated on the server:
+
+```jsx
+function CurrentUserIDSubscription() {
+  const [currentUserID, setCurrentUserID] = useRecoilState(currentUserIDState);
+  let knownServerCurrentUserID;
+
+  // Subscribe server changes to update atom state
+  useEffect(() => {
+    function handleUserChange(id) {
+      knownServerCurrentUserID = id;
+      setCurrentUserID(id);
+    }
+
+    RemoteStateAPI.subscribeToCurrentUserID(handleUserChange);
+    // Specify how to cleanup after this effect
+    return function cleanup() {
+      RemoteServerAPI.unsubscribeFromFriendStatus(handleUserChange);
+    };
+  }, []);
+
+  // Subscribe atom changes to update server state
+  useEffect(() => {
+    if (currentUserID !== knownServerCurrentUserID) {
+      knownServerCurrentID = currentUserID;
+      RemoteServerAPI.updateCurrentUser(currentUserID);
+    }
+  }, [currentUserID, knownServerCurrentUserID]);
+
+  return null;
+}
+```
+
+## Synching State with Parameters
+
+You can also use the [`atomFamily`](/docs/api-reference/utils/atomFamily) helper to sync local state based on parameters.
+
+```jsx
+const friendStatusState = atomFamily({
+  key: 'Friend Status',
+  default: 'offline',
+});
+
+function useFriendStatusSubscription(id) {
+  const setStatus = useSetRecoilState(friendStatusState(id));
+
+  useEffect(() => {
+    RemoteStateAPI.subscribeToFriendStatus(id, setStatus);
```

Comment:
Is it worth making it explicitly clear that it should be taken into consideration that if this hook is used in multiple places, each instance will separately create a subscription?

Reply:
Yeah, I can add a note.  So far, syncing remote mutable state via Recoil isn't a common use-case internally, so we can do some more iteration to provide a more robust example/approach for this use-case.  It would be nice to incorporate more of a singleton pattern here.  But, as also mentioned, lifetimes of components using a friend directly may not match lifetimes of selectors using a friend.  The cleaner answer here may also be related to async query cancelation and memory management for atom/selector lifetime.  If we have an API for that we could perhaps use it for this as well.